src/pages/Compare/index.spec.tsx

```diff
@@ -886,6 +916,192 @@ describe(__filename, () => {
+      moreSetup = (store: Store<ApplicationState>) => {},
+      nextModifiedFile = 'modifiedFile.json',
+    }) => {
+      const version = createExternalVersionWithEntries(
+        [
+          { path: currentFile, status: 'M' },
+          { path: 'fileWithoutChange', status: '' },
+          { path: nextModifiedFile, status: 'M' },
+        ],
+        { id: headVersionId, selected_file: currentFile },
+      );
+      const {
+        _fetchDiff,
+        _fetchVersionFile,
+        dispatchSpy,
+        fakeThunk,
+      } = loadDiffAndRender({
+        addonId,
+        baseVersionId,
+        buildTree,
+        headVersionId,
+        loadDiff,
+        loadVersionFile,
+        moreSetup,
+        version,
+      });
+      return {
+        _fetchDiff,
+        _fetchVersionFile,
+        dispatchSpy,
+        fakeThunk,
+      };
+    };
+
+    it('dispatches fetchDiff and fetchVersionFile for the next file with diff', () => {
+      const addonId = 10;
+      const baseVersionId = 44;
+      const headVersionId = 55;
+      const currentFile = 'current.json';
+      const nextModifiedFile = 'modified.json';
+      const {
+        _fetchDiff,
+        _fetchVersionFile,
+        dispatchSpy,
+        fakeThunk,
+      } = setUpForPreloadAndRender({
+        addonId,
+        baseVersionId,
+        currentFile,
+        headVersionId,
+        nextModifiedFile,
+      });
+
+      expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+      expect(_fetchDiff).toHaveBeenCalledWith({
+        addonId,
+        baseVersionId,
+        headVersionId,
+        path: nextModifiedFile,
+      });
+      expect(_fetchVersionFile).toHaveBeenCalledWith({
+        addonId,
+        versionId: headVersionId,
+        path: nextModifiedFile,
+      });
+    });
+
+    it.each([['buildTree'], ['loadVersionFile'], ['loadDiff']])(
+      'does not dispatch fetchDiff or fetchVersionFile for the next file before %s is dispatched',
+      (action) => {
+        const currentFile = 'current.json';
+        const nextModifiedFile = 'next.json';
+        const { _fetchDiff, _fetchVersionFile } = setUpForPreloadAndRender({
+          currentFile,
+          nextModifiedFile,
+          [action]: false,
+        });
+
+        expect(_fetchDiff).not.toHaveBeenCalledWith(
+          expect.objectContaining({
+            path: nextModifiedFile,
+          }),
+        );
+        expect(_fetchVersionFile).not.toHaveBeenCalledWith(
+          expect.objectContaining(
+            expect.objectContaining({
+              path: nextModifiedFile,
+            }),
+          ),
+        );
+      },
+    );
+
+    it('does not dispatch fetchVersionFile for the next file if it is loading', () => {
+      const headVersionId = 2222;
+      const baseVersionId = 11;
+      const currentFile = 'current.json';
+      const nextModifiedFile = 'next.json';
+      const moreSetup = (store: Store<ApplicationState>) => {
```

Comment:
Instead of passing a callback, I suggest trying to pre-populate a store before handing it to the helper, like this:

```diff
diff --git a/src/pages/Compare/index.spec.tsx b/src/pages/Compare/index.spec.tsx
index e8dfc2c..7c1661b 100644
--- a/src/pages/Compare/index.spec.tsx
+++ b/src/pages/Compare/index.spec.tsx
@@ -928,6 +928,7 @@ describe(__filename, () => {
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       moreSetup = (store: Store<ApplicationState>) => {},
       nextModifiedFile = 'modifiedFile.json',
+      ...params
     }) => {
       const version = createExternalVersionWithEntries(
         [
@@ -951,6 +952,7 @@ describe(__filename, () => {
         loadVersionFile,
         moreSetup,
         version,
+        ...params,
       });
       return {
         _fetchDiff,
@@ -1024,20 +1026,19 @@ describe(__filename, () => {
       const baseVersionId = 11;
       const currentFile = 'current.json';
       const nextModifiedFile = 'next.json';
-      const moreSetup = (store: Store<ApplicationState>) => {
-        store.dispatch(
-          versionsActions.beginFetchVersionFile({
-            path: nextModifiedFile,
-            versionId: headVersionId,
-          }),
-        );
-      };
+      const store = configureStore();
+      store.dispatch(
+        versionsActions.beginFetchVersionFile({
+          path: nextModifiedFile,
+          versionId: headVersionId,
+        }),
+      );
       const { _fetchVersionFile } = setUpForPreloadAndRender({
         baseVersionId,
         currentFile,
         headVersionId,
-        moreSetup,
         nextModifiedFile,
+        store,
       });
 
       expect(_fetchVersionFile).not.toHaveBeenCalledWith(
```

Reply:
I changed it to use parameter `store`.